test/functional/testSettings.js

```diff
@@ -0,0 +1,703 @@
+      // that we will use for all the tests
+      setup: function () {
+        common = new Common(this.remote);
+        settingsPage = new SettingsPage(this.remote);
+        headerPage = new HeaderPage(this.remote);
+        scenarioManager = new ScenarioManager(url.format(config.elasticsearch));
+        remote = this.remote;
+      },
+
+      beforeEach: function () {
+
+        // start each test with an empty kibana index
+        return scenarioManager
+          .unload('emptyKibana')
+          .then(function () {
+            return common
+              .sleep(1000);
+          })
+          .then(function () {
+            return scenarioManager
+              .load('emptyKibana');
+          })
+          .then(function () {
+            return common
+              .sleep(2000);
+          })
+          // and load a minimal set of makelogs data
+          .then(function loadIfEmptyMakelogs() {
+            return scenarioManager
+              .loadIfEmpty('makelogs');
+          })
+          .then(function () {
+            return common.tryForTime(25000, function () {
+              return remote
+                .get(
+                  url.format(_.assign(config.kibana, {
+                    pathname: ''
+                  })))
+                .then(function () {
+                  return common
+                    .sleep(2000);
+                })
+                .then(function () {
+                  return remote
+                    .getCurrentUrl()
+                    .then(function (currentUrl) {
+                      expect(currentUrl).to.contain('settings');
+                    });
+                });
+            });
+          });
+      },
+
+      teardown: function unloadMakelogs() {
+        return scenarioManager
+          .unload('makelogs');
+      },
+
+      /*
+       ** Test the default state of checboxes and the 2 text input fields
+       */
+      'testSettingsInitialState': function () {
+        var testSubName = 'testSettingsInitialState';
+
+        return settingsPage
+          .getTimeBasedEventsCheckbox()
+          .isSelected()
+          .then(function (selected) {
+            expect(selected).to.be.ok();
+          })
+          .then(function () {
+            return settingsPage
+              .getNameIsPatternCheckbox()
+              .isSelected()
+              .then(function (nameIsPatternSelected) {
+                expect(nameIsPatternSelected).to.not.be.ok();
+              });
+          })
+          .then(function () {
+            return settingsPage
+              .getIndexPatternField()
+              .then(function (patternField) {
+                return patternField
+                  .getProperty('value')
+                  .then(function (pattern) {
+                    common.log('patternField value = ' + pattern);
+                    expect(pattern).to.be('logstash-*');
+                  });
+              });
+          })
```

Comment:
Are some of these `then`s deeper than they need to be? Will this work:

```javascript
.then(function () {
  return settingsPage.getIndexPatternField()
})
.then(function (patternField) {
  return patternField.getProperty('value')
})
.then(function (pattern) {
  common.log('patternField value = ' + pattern);
  expect(pattern).to.be('logstash-*');
});
```

It feels easier to read. I haven't tested running this, so might be a subtle promise thingy I'm missing.

And going this route we could add some additional ES6 niceness:

```javascript
.then(() => settingsPage.getIndexPatternField())
.then(patternField => patternField.getProperty('value'));
.then(pattern => {
  common.log('patternField value = ' + pattern);
  expect(pattern).to.be('logstash-*');
});
```